frontend/file/src/main/java/com/virtuslab/gitmachete/frontend/file/highlighting/MacheteAnnotator.java

```diff
@@ -80,6 +87,38 @@ private void processMacheteGeneratedEntry(MacheteGeneratedEntry macheteEntry, An
                   .format(processedBranchName))
           .range(branch).create();
     }
+
+    // update the state of the .git/machete VirtualFile so that new entry is available in the VirtualFile
```

Comment:
Hmmm but note that the lambda passed as param to `invokeLaterIfNeeded` executes on **UI thread**, while the entire `processMacheteGeneratedEntry` method executes on a different, **non-UI thread**.

This means that there's **no** guarantee that at the point when you get to `VfsUtil.loadText(file.getVirtualFile());`, `saveDocumentBeforeCheck(file)` is already complete. UI thread might be busy with other operations, and it might take while for the execution of `saveDocumentBeforeCheck(file)` to start 🤔

Reply:
I don't have a clear idea yet how to properly do this stuff... how about moving the entire logic in `try { ... } catch { ... }` to `saveDocumentBeforeCheck()`, so that it executes only **after** `fileDocManager.saveDocument(...)` is complete, on the same (UI) thread? 🤔

Reply:
Ofc then `saveDocumentBeforeCheck()` would need to be renamed adequately

Reply:
Ok, I tried it. Unfortunately, it does not work. If we add both saving file and performing the check in the 
`ModalityUiUtil.invokeLaterIfNeeded(NON_MODAL, () -> ...`
block - then duplicate entries are not caught for some reason. Let's leave it as it is IMO. It works well.

Reply:
Okay, let's leave as is... maybe just add a comment about the potential caveat (as explained above in https://github.com/VirtusLab/git-machete-intellij-plugin/pull/1216/files#r989323712) in case we observe any race conditions in the future